### PR TITLE
Qy GUI branch

### LINKED ISSUE #116: Modify GUI layout

### PR DESCRIPTION
**Pull request by QY-H00**
1. Modifies the layout of GUI including module list and pie chart of workload distrubution.
2. Modifies the moduleManager to handle workload distribution
3. Need to imporve code quality of moduleManager later.

### CODE REVIEW

**Review by yhtMinceraft1010X — COMMENTED** [review]
Excellent removal of Module's dependency on Workload. LGTM

File: src/main/java/seedu/module/model/ModuleManager.java [hunk 1]
```diff
@@ -53,22 +63,68 @@ public static boolean moduleIsValid(String module) {
...
+     *
+     * @param module the corresponding module of the task
+     * @param task the task that need to be inserted into mapping book.
+     */
+    public static void putIntoCorrectWorkloadDistribution(Module module, Task task) {
+        int workloadLevel = task.getWorkload().getWorkloadLevel();
+        moduleWorkLoadDistribution.put(module,
+            moduleWorkLoadDistribution.get(module) + workloadLevel);
+        switch(workloadLevel) {
+        case 1:
```
Comment (umergta) [inline src/main/java/seedu/module/model/ModuleManager.java]: just a minor suggestion, to change 1,2,3 to low,medium, high for clearer communication through code

File: src/main/java/seedu/module/model/ModuleManager.java [hunk 2]
```diff
@@ -117,8 +182,41 @@ public static void clearMapping() {
...
+    }
+
+    /**
+     * Fills the observable list of existringModules with the keySet of the mapping.
+     */
+    public static void setModulePieChartData() {
+        modulePieChartData.clear();
+        for (int i = 0; i < moduleList.size(); i++) {
+            Module module = moduleList.get(i);
+            modulePieChartData.add(new PieChart.Data(module.toString(), moduleWorkLoadDistribution.get(module)));
```
Comment (umergta) [inline src/main/java/seedu/module/model/ModuleManager.java:208]: Minor suggestion: can use for-each loop instead but no biggie

File: src/main/java/seedu/module/ui/ModuleListPanel.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import javafx.scene.layout.Region;
+import seedu.module.commons.core.LogsCenter;
+import seedu.module.model.task.Module;
+
+/**
+ * Panel containing the list of tasks.
+ */
+public class ModuleListPanel extends UiPart<Region> {
+    private static final String FXML = "ModuleListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(ModuleListPanel.class);
```
Comment (umergta) [inline src/main/java/seedu/module/ui/ModuleListPanel.java:18]: nice use!

File: src/main/java/seedu/module/ui/StatsPanel.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package seedu.module.ui;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.geometry.Side;
+import javafx.scene.chart.PieChart;
```
Comment (umergta) [inline src/main/java/seedu/module/ui/StatsPanel.java:7]: wah didnt know this existed haha

File: src/test/java/seedu/module/model/task/ModuleTest.java [hunk 5]
```diff
@@ -44,67 +43,6 @@ public void isValidModule() {
...
-
-        originalModule.incrementWorkload(new Workload("3"));
-        expectedMessage = getWorkloadCountBuilder(0, 0, 2);
-        assertEquals(originalModule.getWorkloadCount(), expectedMessage);
-
-        originalModule.incrementWorkload(new Workload("2"));
-        originalModule.incrementWorkload(new Workload("1"));
-        expectedMessage = getWorkloadCountBuilder(1, 1, 2);
-        assertEquals(originalModule.getWorkloadCount(), expectedMessage);
-    }
```
Comment (umergta) [inline src/test/java/seedu/module/model/task/ModuleTest.java:70]: think would need to add tests in ModuleManager instead later for the removal of these later

**Review by umergta — COMMENTED** [review]
LGTM! just a few (optional) minor suggestions. and the need to add tests later on maybe in 1.4

File: src/main/java/seedu/module/model/ModuleManager.java (hunk 2 above)
Comment (QY-H00) [reply src/main/java/seedu/module/model/ModuleManager.java:208]: okay! Do you think this is better?
```Java
modulePieChartData.setAll(moduleList.stream().map(module -> new PieChart.Data(module.toString(),
            moduleWorkLoadDistribution.get(module))).collect(Collectors.toList()))
```

File: src/main/java/seedu/module/model/ModuleManager.java (hunk 1 above)
Comment (QY-H00) [reply src/main/java/seedu/module/model/ModuleManager.java]: Resolved.

File: src/main/java/seedu/module/model/ModuleManager.java (hunk 2 above)
Comment (yhtMinceraft1010X) [reply src/main/java/seedu/module/model/ModuleManager.java:208]: If it works, I guess it can be used

**Review by yhtMinceraft1010X — COMMENTED** [review]
LGTM

**Review by AiwassPrime — COMMENTED** [review]
LGTM

**Pull request merged**